color/src/main/java/com/afollestad/materialdialogs/color/DialogColorChooserExt.kt

```diff
@@ -72,3 +93,135 @@ fun MaterialDialog.colorChooser(
+  val tvRedValue: TextView = customPage.findViewById(R.id.tv_red_value)
+  val tvGreenValue: TextView = customPage.findViewById(R.id.tv_green_value)
+  val tvBlueValue: TextView = customPage.findViewById(R.id.tv_blue_value)
+
+  sbAlpha.tint(Color.BLACK)
+  sbRed.tint(Color.RED)
+  sbGreen.tint(Color.GREEN)
+  sbBlue.tint(Color.BLUE)
+
+  initialSelection?.let {
+    if (supportCustomAlpha) {
+      sbAlpha.progress = Color.alpha(it)
+    }
+    sbRed.progress = Color.red(it)
+    sbGreen.progress = Color.green(it)
+    sbBlue.progress = Color.blue(it)
+  } ?: run {
+    sbAlpha.progress = 255
+  }
+
+  llAlpha.visibility = if (supportCustomAlpha) View.VISIBLE else View.GONE
+
+  val listener = object: SeekBar.OnSeekBarChangeListener {
+    override fun onProgressChanged(p0: SeekBar?, p1: Int, p2: Boolean) {
+      onCustomValueChanged(dialog, supportCustomAlpha, waitForPositiveButton, true, customPage, vColor, sbAlpha, sbRed, sbGreen, sbBlue, tvAlphaValue, tvRedValue, tvGreenValue, tvBlueValue, selection)
+    }
+
+    override fun onStartTrackingTouch(p0: SeekBar?) {}
+    override fun onStopTrackingTouch(p0: SeekBar?) { }
+  }
+
+  sbAlpha.setOnSeekBarChangeListener(listener)
+  sbRed.setOnSeekBarChangeListener(listener)
+  sbGreen.setOnSeekBarChangeListener(listener)
+  sbBlue.setOnSeekBarChangeListener(listener)
+
+  onCustomValueChanged(dialog, supportCustomAlpha, waitForPositiveButton, initialSelection != null, customPage, vColor, sbAlpha, sbRed, sbGreen, sbBlue, tvAlphaValue, tvRedValue, tvGreenValue, tvBlueValue, selection)
+}
+
+private fun onCustomValueChanged(
+  dialog: MaterialDialog,
+  supportCustomAlpha: Boolean,
+  waitForPositiveButton: Boolean,
+  valueChanged: Boolean,
+  customView: View,
+  vColor: View,
+  sbAlpha: SeekBar, sbRed: SeekBar, sbGreen: SeekBar, sbBlue: SeekBar,
+  tvAlphaValue: TextView, tvRedValue: TextView, tvGreenValue: TextView, tvBlueValue: TextView,
+  selection: ColorCallback) {
+  if (supportCustomAlpha) {
+    tvAlphaValue.text = sbAlpha.progress.toString()
+  }
+  tvRedValue.text = sbRed.progress.toString()
+  tvGreenValue.text = sbGreen.progress.toString()
+  tvBlueValue.text = sbBlue.progress.toString()
+
+  val color = Color.argb(if (supportCustomAlpha) sbAlpha.progress else 255, sbRed.progress, sbGreen.progress, sbBlue.progress)
+  vColor.setBackgroundColor(color)
+  // simple solution - we save the color as view tag
+  if (valueChanged) {
+    customView.setTag(color)
+    dialog.setActionButtonEnabled(POSITIVE, true)
+  }
+
+  if (!waitForPositiveButton && valueChanged) {
+    selection?.invoke(dialog, color)
+  }
+}
+
+// ----------------
+// Helper functions
+// ----------------
+
+private fun selectedColor(dialog: MaterialDialog, allowCustomColor: Boolean) : Int? {
+  if (allowCustomColor) {
+    val viewPager = dialog.getPager()
+    if (viewPager.currentItem == 1) {
+      return dialog.getPageCustomView().getTag() as Int?
```

Comment:
You want to move the `?` to `as?`, that way an exception won't be thrown if the tag isn't an integer